src/Avalonia.Base/Threading/DispatcherOperation.cs

```diff
@@ -52,6 +53,7 @@ private protected DispatcherOperation(Dispatcher dispatcher, DispatcherPriority
         ThrowOnUiThread = throwOnUiThread;
         Priority = priority;
         Dispatcher = dispatcher;
+        _executionContext = ExecutionContext.Capture();
```

Comment:
WPF checks for ExecutionContext.IsFlowSuppressed() - https://github.com/dotnet/wpf/blob/main/src/Microsoft.DotNet.Wpf/src/Shared/MS/Internal/CulturePreservingExecutionContext.cs#L85